packages/clay-color-picker/src/index.tsx

```diff
@@ -280,44 +290,43 @@ const ClayColorPicker: React.FunctionComponent<IProps> = ({
 									value,
 								])}
 								disabled={disabled}
-								insetBefore
+								insetBefore={!!isHex}
 								onBlur={(event) => {
-									const newColor = tinycolor(
-										event.target.value
-									);
+									if (!isColorValid(event.target.value)) {
+										setInvalidColor(true);
+									}
 
-									const hexString = newColor.isValid()
-										? newColor.toHex()
-										: event.target.value;
+									onValueChange(event.target.value);
 
-									onValueChange(hexString);
-									setHexInputValue(hexString);
 									onBlur(event);
 								}}
 								onChange={(event) => {
-									const newHexValue = event.target.value;
-
-									const newColor = tinycolor(newHexValue);
-
-									setHexInputValue(newHexValue);
-
-									if (newColor.isValid()) {
-										onValueChange(newColor.toHex());
+									if (!isColorValid(event.target.value)) {
+										setInvalidColor(false);
 									}
+
+									onValueChange(event.target.value);
 								}}
 								ref={inputRef}
 								type="text"
-								value={hexInputValue
-									.toUpperCase()
-									.substring(0, 6)}
+								value={value}
 							/>
-
-							<ClayInput.GroupInsetItem before tag="label">
-								{'#'}
-							</ClayInput.GroupInsetItem>
+							{isHex && (
+								<ClayInput.GroupInsetItem before tag="label">
+									{'#'}
+								</ClayInput.GroupInsetItem>
+							)}
```

Comment:
It would be interesting to have some other symbol for the other cases so that there are no blinks to remove the element and then put it back on when you keep switching.

Reply:
Good idea, ill check out other icon examples